### PR TITLE
Issue20877 caf new ztest

### PR DESCRIPTION
**Pull request by zycz**

### CODE REVIEW

File: tests/subsys/caf/sensor_manager/src/main.c [hunk 1]
```diff
@@ -48,60 +42,75 @@ static void test_start(enum test_id test_id)
...
+	event_sensor1->descr = "Simulated sensor 1";
+	APP_EVENT_SUBMIT(event_sensor1);
+
+	event_sensor2->sampling_period = SAMPLING_PERIOD_LONG;
+	event_sensor2->descr = "Simulated sensor 2";
+	APP_EVENT_SUBMIT(event_sensor2);
+
+	event_sensor3->sampling_period = SAMPLING_PERIOD_LONG;
+	event_sensor3->descr = "Simulated sensor 3";
+	APP_EVENT_SUBMIT(event_sensor3);
```
Comment (rakons) [inline tests/subsys/caf/sensor_manager/src/main.c:69]: Make sure that test initialization is finished. For that:
- add event: test_initialization_done and send it here.
- In event processing - when test_initialization_done is received - set the semaphore.
- Here - wait for this semaphore (remember to check the waiting result).

This way we would be sure that no timing issue jumps in and when the test execution is started, all the configuration events are already processed.

**Review by rakons — CHANGES_REQUESTED** [review]

**Review by MarekPieta — COMMENTED** [review]
Seems ok (you even made unit tests independent on each other :) ).
Please just apply @rakons suggestion to ensure that `test_before` takes effect before test is started

**Review by rakons — APPROVED** [review]

**Review by MarekPieta — APPROVED** [review]
Looks good. Thanks for clarification during the offline discussion

**Review by gopiotr — APPROVED** [review]
LGTM. I tested it locally on hardware and it works properly. Thank you a lot for your effort.

**Pull request merged**